target/linux/ar71xx/image/tp-link.mk

```diff
@@ -880,6 +880,21 @@ define Device/tl-wr1043nd-v4
 endef
 TARGET_DEVICES += tl-wr1043nd-v1 tl-wr1043nd-v2 tl-wr1043nd-v3 tl-wr1043nd-v4
 
+define Device/tl-wr942n-v1
+  DEVICE_TITLE := TP-LINK TL-WR942N v1
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
+  BOARDNAME := TL-WR942N-V1
+  TPLINK_BOARD_NAME := TL-WR942N
+  DEVICE_PROFILE := TLWR942NV1
+  IMAGE_SIZE := 14464k
+  KERNEL := kernel-bin | patch-cmdline | lzma | uImageArcher lzma
+  IMAGES := sysupgrade.bin factory.bin
+  IMAGE/sysupgrade.bin := append-rootfs | tplink-safeloader sysupgrade
+  IMAGE/factory.bin := append-rootfs | tplink-safeloader factory
+  MTDPARTS := spi0.0:128k(u-boot)ro,1344k(kernel),13120k(rootfs),64k(product-info)ro,64k(partition-table)ro,64k(soft-version)ro,64k(support-list)ro,640k(config)ro,576k(certificate)ro,64k(usb-config)ro,128k(log)ro,64k(radio-bk)ro,64k(ART)ro,14464k@0x20000(firmware)
```

Comment:
> fwup-ptn os-image base 0x10b35 size 0x10776e
fwup-ptn file-system base 0x1182a3 size 0x978679
fwup-ptn product-info base 0xa9091c size 0x0009f
fwup-ptn partition-table base 0x00800 size 0x00800
fwup-ptn soft-version base 0xa909bb size 0x00015
fwup-ptn support-list base 0xa909d0 size 0x00094
fwup-ptn profile base 0xa90a64 size 0x02658
fwup-ptn default-config base 0xa930bc size 0x0254d

I think we should have at least five last partitions rw as they are always updated during oem updates to have posibility to revert back to any oem firmware, leave partition _product-info_ ro (it contains MAC and PIN) and combine all space between _default-config_ and _ART_ into one another ro partition.